packages/botonic-plugin-watson/src/index.js

```diff
@@ -1,36 +1,99 @@
-import AssistantV1 from 'ibm-watson/assistant/v1'
-import IamAuthenticator from 'ibm-watson/auth'
+import axios from 'axios'
+
+import WatsonOutputParser from './watson-output-parser'
 
 export default class BotonicPluginWatson {
+  REQUIRED_OPTION_FIELDS = ['apikey', 'url', 'assistant_id']
+
+  EXPECTED_SESSION_ID_REQUEST_STATUS = 201
+  EXPECTED_OUTPUT_REQUEST_STATUS = 200
+
   constructor(options) {
-    this.options = options
+    try {
+      this.validateOptions(options)
+      this.init(options)
+    } catch (e) {
+      console.error(String(e))
+    }
+  }
+
+  validateOptions(options) {
+    if (!options) throw new Error('No options provided.')
+    if (!this.areRequiredOptionsProvided(options))
+      throw new Error('Missing one or more fields.')
```

Comment:
nice that you check the options are provided.
But think on the user (developer). If you don't provide which one is missing, it will be difficult to find which one is wrong (maybe there's a misspell, maybe a bug lead to a null value...)
I suggest this approach
* here you call "Object.keys(o => checkRequiredOption(o))"
```
function  checkRequiredOption(key) {
  if key does not exist {
   throw "missing option ${key}
  }
  if option[key] == undefined {
   throw "undefined value for option ${key}"
  }
}
```